### DISCUSSION TITLE
Hidden dropdown

### DISCUSSION THREAD

**Discussion by alekseykurylev** [ariakit/ariakit#3320] [Questions] [answered]
Help! I can't find which props is responsible for hiding the dropdown area if I remove the cursor.

https://github.com/ariakit/ariakit/assets/8526127/fe035a3d-921b-4151-8fa2-4210dd9af6d7

**Comment by diegohaz** [comment]
[Hovercard](https://ariakit.org/components/hovercard), used by submenus, will automatically hide when hovering outside. However, it won't hide if there's DOM focus within it.

In that example, the [`focusOnHover`](https://ariakit.org/reference/menu-item#focusonhover) prop is disabled on the menu items.

**Reply by alekseykurylev** [reply]
@diegohaz  When I specify "focusOnHover", dropdown closes instantly. How can I add a hide delay?

**Reply by diegohaz** [reply]
Could you please create a reproduction in CodeSandbox or StackBlitz? The hovercard shouldn't hide instantly when you're moving the mouse toward it.

**Reply by alekseykurylev** [reply]
@diegohaz  [codesandbox](https://codesandbox.io/p/sandbox/menubar-2g74tg?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clr9asqzd00063v6mxhm73cy3%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clr9asqzd00023v6mx88w86x2%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clr9asqzd00033v6mtt0e04oj%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clr9asqzd00053v6mh13yto2s%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clr9asqzd00023v6mx88w86x2%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr9asqzc00013v6mhl6uaaoe%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpublic%252Findex.html%2522%257D%255D%252C%2522id%2522%253A%2522clr9asqzd00023v6mx88w86x2%2522%252C%2522activeTabId%2522%253A%2522clr9asqzc00013v6mhl6uaaoe%2522%257D%252C%2522clr9asqzd00053v6mh13yto2s%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr9asqzd00043v6m4um9qy9v%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clr9asqzd00053v6mh13yto2s%2522%252C%2522activeTabId%2522%253A%2522clr9asqzd00043v6m4um9qy9v%2522%257D%252C%2522clr9asqzd00033v6mtt0e04oj%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clr9asqzd00033v6mtt0e04oj%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Reply by alekseykurylev** [reply]
I am using the Menubar component with a submenu display when hover over the cursor.

**Reply by diegohaz** [reply, marked as answer]
Got it. I had thought you mentioned the dropdown was closing immediately when you tried to move the mouse toward it, which would be a bug.

Use the [`hideTimeout`](https://ariakit.org/reference/menu-provider#hidetimeout) prop.

**Reply by alekseykurylev** [reply]
Thanks

**Reply by alekseykurylev** [reply]
@diegohaz  I noticed one problem, if you click on the item in the drop-down window, then the next time you open the drop-down menu, closing it does not work if you remove the cursor.

https://github.com/ariakit/ariakit/assets/8526127/38646f0d-a3b0-4851-9b8c-ac3d42f047b0
